src/commands/birthday/_common.ts

```diff
@@ -0,0 +1,140 @@
+import { Cursor, Collection as MongoCollection } from 'mongodb';
+import ExecutionContext from '../../ExecutionContext';
+import * as Discord from '../../Discord';
+import { Collection } from '../../database';
+
+export type Birthday = {
+  _id: number;
+  users: string[];
+};
+
+/**
+ * Calculate the day of year from a date.
+ *
+ * @param date The date to convert.
+ */
+export const calculateDayOfYear = (date: string): number => {
+  const now = new Date(date);
+  const start = new Date(now.getFullYear(), 0, 0);
+  const diff =
+    now.getTime() -
+    start.getTime() +
+    (start.getTimezoneOffset() - now.getTimezoneOffset()) * 60 * 1000;
+  const oneDay = 1000 * 60 * 60 * 24;
+  const dayOfYear = Math.floor(diff / oneDay);
+
+  return dayOfYear;
+};
+
+/**
+ * Calculate a day of year to a Date object.
+ *
+ * @param day THe day of year to convert.
+ */
+export const calculateDate = (day: number): Date => {
+  const startOfCurrentYear = new Date(new Date().getFullYear(), 0); // Initialize a date to Jan. 1 of the current year
+  const calculatedDate = new Date(startOfCurrentYear.setDate(day)); // Add the number of days.
+
+  return calculatedDate;
+};
+
+export const getTargetUser = async (
+  ctx: ExecutionContext,
+  requestorId: string | undefined,
+  targetUserId: string,
+): Promise<Discord.User | undefined> => {
+  if (targetUserId) {
+    return await ctx.api.getUser(targetUserId);
+  } else if (requestorId) {
+    return await ctx.api.getUser(requestorId);
+  }
+};
+
+export const dayExists = async (
+  ctx: ExecutionContext,
+  guildId: string,
+  day: number,
+): Promise<boolean> => {
+  return (
+    (await birthdaysCollection(ctx, guildId).findOne({ _id: day })) !== null
+  );
+};
+
+export const userHasBirthday = async (
+  ctx: ExecutionContext,
+  guildId: string,
+  userId: string,
+): Promise<boolean> => {
+  return (
+    (await birthdaysCollection(ctx, guildId).findOne({ users: userId })) !==
+    null
+  );
+};
+
+export const setBirthday = async (
+  ctx: ExecutionContext,
+  guildId: string,
+  dayOfYear: number,
+  userId: string,
+): Promise<any> => {
+  const day = await dayExists(ctx, guildId, dayOfYear);
+
+  if (day) {
```

Comment:
Not necessarily something that needs to be done here, but it could be nice in the future to make a `createIfNotExists` or something that updates a document if it is present, or makes a new one if not to simplify this step. Here I think it's fine though.